### PR TITLE
Add traffic prediction example

### PR DESCRIPTION
**Pull request by aurorarossi**
This PR adds an example of the use of a temporal graph neural network for traffic prediction.
(Before merging this, the following PRs must be merged: https://github.com/CarloLucibello/GraphNeuralNetworks.jl/pull/319 and https://github.com/JuliaML/MLDatasets.jl/pull/216)

### CODE REVIEW

**Comment by CarloLucibello** [comment]
Did you run the example on gpu as well?

**Comment by aurorarossi** [comment]
Yes!

File: examples/traffic_prediction.jl [hunk 1]
```diff
@@ -0,0 +1,96 @@
+# Temporal graph neural networks for traffic prediction
```
Comment (CarloLucibello) [inline examples/traffic_prediction.jl:1]: please expand a bit this comment with the description of the example and add bibliographic references

File: examples/traffic_prediction.jl [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+function getdataset()
+    metrla = METRLA(; num_timesteps = 3)
+    g=metrla[1]
+    features=[]
+    targets=[]
+    graph = GNNGraph(g.edge_index; edata = g.edge_data, g.num_nodes)
+    for i in 1:length(g.node_data.features)
+        push!(features, g.node_data.features[i])
+        push!(targets,g.node_data.targets[i])
+    end
```
Comment (CarloLucibello) [inline examples/traffic_prediction.jl:22]:
```suggestion
    features = [x for x in g.node_data.features]
    targets = [y for y in g.node_data.targets]
```
or just directly `features = g.node_data.features`  if that's possible

**Pull request merged**